physics/forkable_body.hpp

```diff
@@ -110,19 +110,17 @@ ForkableIterator<Tr4jectory, It3rator, Traits>::current() const {
 template<typename Tr4jectory, typename It3rator, typename Traits>
 void ForkableIterator<Tr4jectory, It3rator, Traits>::NormalizeIfEnd() {
   CHECK(!ancestry_.empty());
-  if (current_ == ancestry_.front()->timeline_end() &&
-      ancestry_.size() > 1) {
-    ancestry_.erase(ancestry_.begin(), --ancestry_.end());
-    current_ = ancestry_.front()->timeline_end();
+  if (current_ == ancestry_.back()->timeline_end() && ancestry_.size() > 1) {
+    ancestry_ = {ancestry_.front()};
```

Comment:
I'd imagine that `ancestry_.resize(1)` would be more efficient in terms of heap management and a bit more readable IMO.

Reply:
I agree that resize would be more readable, but it doesn't compile since not_null<> has no default constructor.

Reply:
Let’s make it an `erase(ancestry_.begin() + 1, ancestry_.end())` then.